deckhouse-controller/pkg/controller/deckhouse-release/updater/updater.go

```diff
@@ -41,19 +41,19 @@ func NewDeckhouseUpdater(logger logger.Logger, client client.Client, dc dependen
 	podIsReady, clusterBootstrapping bool, imagesRegistry string, enabledModules []string) (*updater.Updater[*v1alpha1.DeckhouseRelease], error) {
 	return updater.NewUpdater[*v1alpha1.DeckhouseRelease](logger, updateSettings.NotificationConfig, updateSettings.Mode, releaseData,
 		podIsReady, clusterBootstrapping, newKubeAPI(client, dc, imagesRegistry),
-		newMetricsUpdater(), newValueSettings(updateSettings.DisruptionApprovalMode), newWebhookDataGetter(), enabledModules), nil
+		newMetricsUpdater(), newValueSettings(updateSettings.DisruptionApprovalMode), newWebhookDataSource(), enabledModules), nil
 }
 
-func newWebhookDataGetter() *webhookDataGetter {
-	return &webhookDataGetter{}
+func newWebhookDataSource() *webhookDataSource {
+	return &webhookDataSource{}
 }
 
-type webhookDataGetter struct {
+type webhookDataSource struct {
 }
 
-func (w *webhookDataGetter) GetMessage(release *v1alpha1.DeckhouseRelease, releaseApplyTime time.Time) string {
-	version := fmt.Sprintf("%d.%d", release.GetVersion().Major(), release.GetVersion().Minor())
-	return fmt.Sprintf("New Deckhouse Release %s is available. Release will be applied at: %s", version, releaseApplyTime.Format(time.RFC850))
+func (webhookDataSource) Fill(output *updater.WebhookData, _ *v1alpha1.DeckhouseRelease, applyTime time.Time) {
```

Comment:
`newWebhookDataSource()` returns a reference to the structure, why then is the Fill method created on the structure and not on the reference? This will result in additional type transformations before the method is called.

I also wonder what happens if we pass nil as a reference to updater.WebhookData in the Fill method. Maybe it is worth checking the moment and avoiding accessing fields of a non-existent structure?

Reply:
Fixed